helm-charts/splunk-otel-collector/values.yaml

```diff
@@ -423,13 +423,65 @@ clusterReceiver:
   annotations: {}
   podAnnotations: {}
 
-  # This flag enables Kubernetes events collection using OpenTelemetry Kubernetes Events Receiver
-  # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8seventsreceiver
+  # This flag enables Kubernetes objects collection using OpenTelemetry Kubernetes Events Receiver
+  # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sobjectsreceiver
   # This option requires `logsEnabled` to be set to `true` for either `splunkObservability` or `splunkPlatform`
-  # depending on where you want to send the events. Otherwise this option will not have any effect.
+  # depending on where you want to send the events. Otherwise, this option will not have any effect.
   # The receiver currently is in alpha state which means that events format might change over time.
   # Once the receiver is stabilized, it'll be enabled by default in this helm chart
-  eventsEnabled: false
+  objectsEnabled: false
+
+  # Define what objects to collect (pull/watch) from the Kubernetes API server. Examples are here:
+  #
+  # == Schema ==
+  # ```
+  # objects:
+  #   <apiGroup>:
+  #     - <objectDefinition>
+  # ```
+  #
+  # Each `objectDefinition` has the following fields:
+  # * mode:
+  #     define in which way it collects this type of object, either "pull" or "watch".
+  #     - "poll" mode will read all objects of this type use the list API at an interval.
+  #     - "watch" mode will setup a long connection using the watch API to just get updates.
+  # * name: [REQUIRED]
+  #     name of the object, e.g. `pods`, `namespaces`.
+  # * namespace:
+  #     only collects objects from the specified namespace, by default it's all namespaces
+  # * labelSelector:
+  #     select objects by label(s)
+  # * fieldSelector:
+  #     select objects by field(s)
+  # * interval:
+  #     the interval at which object is pulled, default 15 minutes.
+  #     Only useful for "pull" mode.
+  #
+  #
+  # == Example ==
+  # ```
+  # objects:
+  #   v1:
+  #     - name: pods
+  #       namespace: default
+  #       mode: pull
+  #       interval: 60m
+  #     - name: events
+  #       mode: watch
+  #   apps:
+  #     - name: daemonsets
+  #       labelSelector: environment=production
+  # ```
+  objects:
+    v1:
+      - name: pods
+        mode: pull
+      - name: namespaces
+        mode: pull
+      - name: nodes
+        mode: pull
+      - name: events
+        mode: watch
```

Comment:
Why do we need this config interface causing all the complicated translations? Why can't we use the same config interface provided by https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sobjectsreceiver?

Reply:
It is required to add rules for all objects in the ClusterRole manifest.
k8sobjectsreceiver automatically detects the resource apiGroups by looking up the resource name from the kube-apiserver. But I don't think that the helm chart provides such functionality.

Reply:
Do we really need the autogeneration of `ClusterRole` manifest based on this config? Introducing another config interface layer that is different from the receiver's one just to build `ClusterRole` with all the helm magic that needs to be maintained seems like an overcomplication to me.

What if we keep `eventsEnabled` and `objectsEnabled` that generate some default config for `k8sobjectsreceiver` with a generic list of the most common objects? And, if users need different behavior, they can reconfigure `clusterReceiver.config.receivers.k8sobjects.objects` and potentially `rbac.customRules`?

Reply:
> Do we really need the autogeneration of ClusterRole manifest based on this config? Introducing another config interface layer that is different from the receiver's one just to build ClusterRole with all the helm magic that needs to be maintained seems like an overcomplication to me.

Makes sense. 

> What if we keep eventsEnabled and objectsEnabled that generate some default config for k8sobjectsreceiver with a generic list of the most common objects? And, if users need different behavior, they can reconfigure clusterReceiver.config.receivers.k8sobjects.objects and potentially rbac.customRules?

Agreed. It will allow us to follow the same config interface used by k8sobjects. 

Do we need both `eventsEnabled` and `objectsEnabled`? k8sobjects can already collect events but with different structures.

Reply:
I agree that we should migrate to `k8sobjects` receiver for events. We can keep the same `eventsEnabled` option that renders part of `k8sobjects` receiver config instead of `k8sevents` receiver. The only concern I have is that this is a breaking change for users, we need to document it and provide a config snippet on how to configure `k8sevents` receiver via `clusterReceiver.config` for users that want to keep the existing behavior.

Reply:
Maybe we can allow both k8sobject and k8s_event, and document that k8s_event will be removed in future and how to migrate.

Reply:
Maybe we can deprecate `eventsEnabled` option in favor of just one `k8sObjects` option that will expose the k8sobjects receiver interface as is. `k8sObjects` default value will be `[]` resulting in the `k8sobjects` receiver not being set up. Deprecating `eventsEnabled`, we will ask the user to setup 

```yaml
  k8sObjects:
    - mode: watch
      name: events
```

instead. It should collect the same data just in a different format. Users can also specify any other objects additionally in `k8sObjects` section, but they need to make sure that `rbac.customRules` is defined when needed. WDYT?

Reply:
Agreed